boxes.cc

```diff
@@ -31,14 +35,30 @@ using namespace benchmark;
 // Boxes:
 // Spawn a single box and record accuracy for momentum and enery
 // conservation
-void BoxesTest::Boxes(const std::string &_physicsEngine
-                    , double _dt
-                    , int _modelCount
-                    , bool _collision
-                    , bool _complex)
-{
+void BoxesTest::Boxes(const std::string &_physicsEngine, double _dt,
+                      int _modelCount, bool _collision, bool _complex) {
+  ASSERT_GT(_modelCount, 0);
+  std::string world_erb_path =
+      boost::str(boost::format("%1%/boxes.world.erb") % WORLDS_DIR_PATH);
+
+  std::string world_path = boost::str(
+      boost::format(
+          "%1%/%2%/"
+          "boxes_collision%3%_complex%4%_dt%5$.0e_modelCount%6%.world") %
+      WORLDS_DIR_PATH % TEST_NAME % _collision % _complex % _dt % _modelCount);
```

Comment:
If not for formatting the `dt` value, I would recommend using a `stringstream` instead of `boost::format` as I find it more readable to have the values expressed inline. It's too bad `boost::format` doesn't use the same [formatting syntax](https://en.cppreference.com/w/cpp/utility/format/spec) as the [c++20 formatting library](https://en.cppreference.com/w/cpp/utility/format).

We intentionally avoid using boost in `gz-*` libraries. Since gazebo-classic is already using boost, I think it's fine here, but we may want to avoid it when integrating with the newer gazebo

Reply:
oh I see, I was using boost because it was already being used by gazebo classic. I will make sure to stick to `stringstream` for newer gazebo.